mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/logging/impl/Log.java

```diff
@@ -75,4 +76,17 @@
 
 	@Message(id = 1219, value = "Database %s does not exist. Either create it yourself or set property '" + OgmProperties.CREATE_DATABASE + "' to true.")
 	HibernateException databaseDoesNotExistException(String databaseName);
+
+	@Message(id = 1220, value = "When using MongoDB it is not valid to use a name for a table (a collection) which starts with the 'system.' prefix."
+			+ " Please change name for '%s', for example using @Table ")
+	MappingException collectionNameHasInvalidSystemPrefix(String qualifiedName);
+
+	@Message(id = 1221, value = "When using MongoDB it is not valid to use a name for a table (a collection) which contains the NUL character '\\0'."
+			+ " Please change name for '%s', for example using @Table ")
+	MappingException collectionNameContainsNULCharacter(String qualifiedName);
+
+	@Message(id = 1222, value = "When using MongoDB it is not valid to use a name for a table (a collection) which contains the dollar character '$'."
```

Comment:
Maybe add a reference to inner classes, so that people understand where the '$' potentially comes from!?